listenbrainz/webserver/views/stats_api.py

```diff
@@ -550,3 +584,29 @@ def _get_entity_list(
     elif entity == 'recording':
         return getattr(stats, stats_range).recordings[offset:count]
     raise APIBadRequest("Unknown entity: %s" % entity)
+
+
+def _get_country_codes_by_msids(artist_msids):
+    country_map = defaultdict(int)
+    for index in range(0, len(artist_msids), 50):
+        current_batch = artist_msids[index: index + 50]
+        try:
+            artist_credit_id_data = requests.get('http://bono.metabrainz.org:8000/artist-msid-lookup/json', {
+                '[artist_msid]': ','.join(current_batch)
+            }).json()
+            artist_credit_ids = [
+                x['artist_credit_id'] for x in artist_credit_id_data]
+
+            location_data = requests.get('http://bono.metabrainz.org:8000/artist-credit-id-country-code/json', {
+                '[artist_credit_id]': ','.join([str(id) for id in artist_credit_ids])
+            }).json()
+            for location in location_data:
+                country = alpha_2_to_3[location['area_code'].split(',')[0]]
+                country_map[country] += 1
+        except json.decoder.JSONDecodeError:
```

Comment:
should probably break this into two, so that we know which request caused the decode error.